src/codegen/signal.rs

```diff
@@ -134,14 +133,13 @@ pub fn generate(
 
             writeln!(
                 w,
-                "{}let {} = unsafe {{ glib::Object::from_glib_borrow(self.as_ptr() as *mut {}).emit_by_name(\"{}\", &[{}]).unwrap() }};",
+                "{}let {} = self.emit_by_name(\"{}\", &[{}]);",
```

Comment:
And one day we also move the signal connection functions over to `ObjectExt::connect()` :)

Reply:
indeed, a task for another day

Reply:
Definitely. It's not so easy :)